syncer/optimist.go

```diff
@@ -100,10 +113,15 @@ func (s *Syncer) handleQueryEventOptimistic(
 	var (
 		rev    int64
 		skipOp bool
+		op     optimism.Operation
 	)
 	switch needTrackDDLs[0].stmt.(type) {
-	case *ast.CreateDatabaseStmt, *ast.DropDatabaseStmt:
-	// for CREATE/DROP DATABASE, we do nothing now.
+	case *ast.CreateDatabaseStmt, *ast.AlterDatabaseStmt:
+		op.DDLs = needHandleDDLs
+		skipOp = true
+	case *ast.DropDatabaseStmt:
+		skipOp = true
+		needHandleDDLs = []string{} // no DDL needs to be handled for `DROP DATABASE` now.
```

Comment:
It seems no need to do this now because we do `needHandleDDLs = op.DDLs` in L160 (the same for L138).